compiler/rustc_codegen_ssa/src/mir/constant.rs

```diff
@@ -38,24 +35,14 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 ),
             },
             mir::ConstantKind::Unevaluated(uv, _) => uv,
-            mir::ConstantKind::Val(val, _) => return Ok(val),
+            mir::ConstantKind::Val(val, _) => return val,
         };
 
-        self.cx.tcx().const_eval_resolve(ty::ParamEnv::reveal_all(), uv, None).map_err(|err| {
-            match err {
-                ErrorHandled::Reported(_) => {
```

Comment:
In fact here we ignored `is_tainted_by_errors` in the old code. So we're now better at honoring `is_tainted_by_errors` than before.

Maybe the problem is that we're also better at keeping the span than before, so the fact that something doesn't get tainted properly is more visible now.